internal/attestation/crafter/crafter.go

```diff
@@ -463,8 +463,32 @@ func (c *Crafter) ResolveEnvVars(ctx context.Context, attestationID string) erro
 	return nil
 }
 
-// Inject material to attestation state
-func (c *Crafter) AddMaterial(ctx context.Context, attestationID, key, value string, casBackend *casclient.CASBackend, runtimeAnnotations map[string]string) error {
+// AddMaterialContractFree adds a material to the crafting state without checking the contract schema.
+// This is useful for adding materials that are not defined in the schema.
+// The name of the material is automatically calculated to conform the API contract.
+func (c *Crafter) AddMaterialContractFree(ctx context.Context, attestationID, kind, value string, casBackend *casclient.CASBackend, runtimeAnnotations map[string]string) error {
+	if err := c.requireStateLoaded(); err != nil {
+		return err
+	}
+
+	// 1 - Try to parse incoming type to a known kind
+	m := schemaapi.CraftingSchema_Material{}
+	if val, found := schemaapi.CraftingSchema_Material_MaterialType_value[kind]; found {
+		m.Type = schemaapi.CraftingSchema_Material_MaterialType(val)
```

Comment:
awesome, glad that you are using this :)

Reply:
Ran into a few problems that I'm glad I finally overcome them :D